.github/workflows/docker-git.yml

```diff
@@ -0,0 +1,43 @@
+# Publishes the Docker image, only to be used with `workflow_dispatch`. The
+# images from this workflow will be tagged with the git sha of the branch used
+# and will NOT tag it as `latest`.
+
+name: docker-git
+
+on:
+  workflow_dispatch: {}
+
+env:
+  REPO_NAME: ${{ github.repository_owner }}/reth
+  IMAGE_NAME: ${{ github.repository_owner }}/reth
+  OP_IMAGE_NAME: ${{ github.repository_owner }}/op-reth
+  CARGO_TERM_COLOR: always
+  DOCKER_IMAGE_NAME: ghcr.io/${{ github.repository_owner }}/reth
+  OP_DOCKER_IMAGE_NAME: ghcr.io/${{ github.repository_owner }}/op-reth
+  DOCKER_USERNAME: ${{ github.actor }}
+
+jobs:
+  build:
+    name: build and push
+    runs-on: ubuntu-20.04
+    permissions:
+      packages: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+      - uses: Swatinem/rust-cache@v2
+        with:
+          cache-on-failure: true
+      - uses: taiki-e/install-action@cross
+      - name: Log in to Docker
+        run: |
+          echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io --username ${DOCKER_USERNAME} --password-stdin
+      - name: Set up Docker builder
+        run: |
+          docker run --privileged --rm tonistiigi/binfmt --install arm64,amd64
+          docker buildx create --use --name cross-builder
+      - name: Build and push the git-tagged reth image
+        run: make PROFILE=maxperf docker-build-push
+      - name: Build and push the git-tagged op-reth image
```

Comment:
where is the image tagged & pushed with the git sha? in the makefile it looks like it uses the latest tag or am I missing something?

Reply:
The `docker-build-push` makefile target uses the `GIT_TAG` env var as an argument:
https://github.com/paradigmxyz/reth/blob/600a394571e8397295256b25ed50c8c9ba6f4c8d/Makefile#L194-L200

Which is set in the makefile like this:
https://github.com/paradigmxyz/reth/blob/600a394571e8397295256b25ed50c8c9ba6f4c8d/Makefile#L4

This then gets passed as the tag here:
https://github.com/paradigmxyz/reth/blob/600a394571e8397295256b25ed50c8c9ba6f4c8d/Makefile#L230-L231

See for contrast the `build-push-latest` makefile target, which uses `latest` as the second argument:
https://github.com/paradigmxyz/reth/blob/600a394571e8397295256b25ed50c8c9ba6f4c8d/Makefile#L202-L208

Reply:
yep makes sense, so it means we first have to push a tag with the git sha for this to work right?

Reply:
> push a tag with the git sha for this to work right?

Actually we will have a dropdown like this:
![Screenshot 2024-10-14 at 4 47 35 PM](https://github.com/user-attachments/assets/45f86d14-16c9-4447-81c3-77648098b45e)

So it should just work for main, and if we want a specific commit, we'll have to push a branch first.

Reply:
oops, I totally misunderstood the code here. Yes that command only gets tags. Let me get something working with git sha

Reply:
It’s possible to get the sha via GitHub actions {{ git.gitsha }}

Reply:
Now this should use the git sha, there is also a new makefile target